### ISSUE TITLE
customize value function in qKG [Feature Request]

### ISSUE THREAD

**Issue by shalijiang** [pytorch/botorch#349]
currently the value function in qKG is either SimpleRegret or PosteriorMean. Why not parameterize this?  So that users can pass in whatever value function of their own design?

**Comment by Balandat** [comment]
Yes, that’s very sensible. We have some dev-level code for this that is part of a larger project to allow multi-step look-ahead acquisition with generic terminal value functions (of which your suggestion is a special case). We’ll clean this up a bit and hopefully we can push this as an experimental feature soon.

**Comment by shalijiang** [comment]
sounds great!
I was thinking about using `qEI` as the value function, which seems easy to support if we remove the exception 
```
        elif not (best_f.ndim < 2 and best_f.numel() == 1):
            raise ValueError(
                "best_f must be either a scalar or a single-elment "
                "one-dimensional tensor."
            )
```
https://github.com/pytorch/botorch/blob/124223491c2ca22b916588b737894eb576c51870/botorch/acquisition/monte_carlo.py#L138
and pass in 
`best_f = fantasy_model.train_targets.max(dim=-1)[0].unsqueeze(-1)`

**Comment by Balandat** [comment]
@shalijiang #357 implements a generic multi-step lookahead strategy for which you can specify the value function (this is kind of naive, and potentially very expensive, in particular in terms of memory.) 

I still need to write unit tests and whip up some basic usage examples before landing this (as an experimental feature), but feel free to play around with it. Very interested in hearing your thoughts on this.

**Comment by Balandat** [comment]
For the record, #410 goes a step further and allows specifying value functions for each look-ahead stage.

**Comment by Balandat** [comment]
That PR is superseded by #611 - hoping to clean this up and get it in soon.

**Comment by Balandat** [comment]
Available now as part of #611
